### DISCUSSION TITLE
Can I hook a request to an internal server?

### DISCUSSION THREAD

**Discussion by mondy** [wailsapp/wails#1262] [Q&A]
I want to dynamically generate images and display them via HTTP (use `img` tag). 
I came up with a way to hook the asset, I can't use the GET parameter.
Do I have to run another server?

**Comment by leaanthony** [comment]
Hi. I've been considering using the asset server as a cache, but am now wondering whether callbacks to Go code would be better. Maybe a handler for certain paths and query Params? Would this be a solution?

**Reply by stffabi** [reply]
What would you think about registering a `http.Handler` on the options? That would make it possible for users to use http routers they are already familiar with and they could also e.g. provide HTTP response headers. Maybe if the assets are seen as cache, first the cache/assets could be queried and if the file is not found, it could fallback to the `http.Handler`. This could also make it easily possible to redirect the files by proxying them to another web server if they want to.
Just an idea which would need more thinking, especially if the processRequest can mapped to a `http.Handler` in a good way.

**Reply by mondy** [reply]
> What would you think about registering a `http.Handler` on the options? 

I think it's good.

**Reply by leaanthony** [reply]
Amazing idea!

**Reply by stffabi** [reply]
Maybe I'll find some spare time this week to provide a draft PR 😄 .

**Reply by stffabi** [reply]
You might be interested in this PR  #1325. Would be awesome if you could give it a try.
